src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -3554,16 +3554,23 @@ def _set_route_table(ncf, resource_group_name, route_table, subnet):
 def create_subnet(cmd, resource_group_name, virtual_network_name, subnet_name,
                   address_prefix, network_security_group=None,
                   route_table=None, service_endpoints=None, service_endpoint_policy=None,
-                  delegations=None):
+                  delegations=None, nat_gateway=None):
     NetworkSecurityGroup, ServiceEndpoint, Subnet, SubResource = cmd.get_models(
         'NetworkSecurityGroup', 'ServiceEndpointPropertiesFormat', 'Subnet', 'SubResource')
     ncf = network_client_factory(cmd.cli_ctx)
+
     if cmd.supported_api_version(min_api='2018-08-01'):
         subnet = Subnet(
             name=subnet_name,
             address_prefixes=address_prefix if len(address_prefix) > 1 else None,
             address_prefix=address_prefix[0] if len(address_prefix) == 1 else None
         )
+        if nat_gateway:
+            subnet.nat_gateway = SubResource(id=resource_id(
+                subscription=get_subscription_id(cmd.cli_ctx),
+                resource_group=resource_group_name,
+                namespace='Microsoft.Network', type='natGateways',
+                name=nat_gateway))
```

Comment:
This should use name or ID logic. https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters